forte/data/readers/classification_reader.py

```diff
@@ -0,0 +1,239 @@
+     title].
+    By checking the actual data, except for label, we must find their corresponding ontology names
+    and its relative path in the given or customized ontology.
+    Here we define data_fields = ["label", "ft.onto.base_ontology.Title",
+     "ft.onto.ag_news.Description"]
+    And we want both Title and Description to be included in input string, so we define
+    subtext fields = [ "ft.onto.base_ontology.Title",  "ft.onto.ag_news.Description"].
+    To see a full example, please refer to
+    https://github.com/asyml/forte/examples/classification
+    """
+
+    def __init__(self):
+        super().__init__()
+        self._index2class = None
+        self._class2index = None
+
+    def set_up(self):
+        assert self.configs.index2class is not None, (
+            "User must set _index2class to enable"
+            " the dataset reader to encode labels correctly"
+        )
+
+        # class and index
+        if not self.configs.digit_label:
+            # initialize class
+            self._index2class = self.configs.index2class
+            self._class2index = {v: k for k, v in self._index2class.items()}
+
+    def initialize(self, resources: Resources, configs: Config):
+        super().initialize(resources, configs)
+        self.set_up()
+        if "label" not in self.configs.data_fields:
+            raise ProcessorConfigError(
+                "There must be data field named 'label' in reader config."
+            )
+
+        if not self.configs.subtext_fields:
+            raise ProcessorConfigError(
+                "There must be at least one subtext field "
+                + "to reader to select from."
+            )
+
+        if not set(self.configs.subtext_fields).issubset(
+            set(self.configs.data_fields)
+        ):
+            raise ProcessorConfigError(
+                "subtext fields must be a subset of data fields"
+            )
+
+    def _collect(self, csv_file: str) -> Iterator[Tuple[int, str]]:
+        with open(csv_file, encoding="utf-8") as f:
+            data = csv.reader(f, delimiter=",", quoting=csv.QUOTE_ALL)
+            if self.configs.skip_first_line:
+                next(data)
+            for line_id, line in enumerate(data):
+                yield line_id, line
+
+    def _cache_key_function(self, line_info: Tuple[int, List[str]]) -> str:
+        return str(line_info[0])
+
+    def _parse_pack(
+        self, line_info: Tuple[int, List[str]]
+    ) -> Iterator[DataPack]:
+        line_id, line = line_info
+        # content label title
+        pack = DataPack()
+
+        # subtext fields must follow the ontology names
+        data_fields = self.configs.data_fields
+
+        if len(self.configs.data_fields) != len(line):
+            raise ProcessorConfigError(
+                "Data fields provided in config "
+                "is not aligned with the actual line info from dataset.\n"
```

Comment:
so what is "actual line info", please use more user friendly terms. I feel like line info means the column format in the dataset.